servers/rendering/rendering_server_default.h

```diff
@@ -134,48 +136,68 @@ class RenderingServerDefault : public RenderingServer {
+			RSG::texture_storage->m_type##_initialize(ret, p1, p2);                                                  \
+		}                                                                                                            \
+		return ret;                                                                                                  \
 	}
 
-#define FUNCRIDTEX6(m_type, m_type1, m_type2, m_type3, m_type4, m_type5, m_type6)                                                \
-	virtual RID m_type##_create(m_type1 p1, m_type2 p2, m_type3 p3, m_type4 p4, m_type5 p5, m_type6 p6) override {               \
-		RID ret = RSG::texture_storage->texture_allocate();                                                                      \
-		if (Thread::get_caller_id() == server_thread || RSG::texture_storage->can_create_resources_async()) {                    \
-			RSG::texture_storage->m_type##_initialize(ret, p1, p2, p3, p4, p5, p6);                                              \
-		} else {                                                                                                                 \
-			command_queue.push(RSG::texture_storage, &RendererTextureStorage::m_type##_initialize, ret, p1, p2, p3, p4, p5, p6); \
-		}                                                                                                                        \
-		return ret;                                                                                                              \
+#define FUNCRIDTEX6(m_type, m_type1, m_type2, m_type3, m_type4, m_type5, m_type6)                                                    \
+	virtual RID m_type##_create(m_type1 p1, m_type2 p2, m_type3 p3, m_type4 p4, m_type5 p5, m_type6 p6) override {                   \
+		RID ret = RSG::texture_storage->texture_allocate();                                                                          \
+		if (create_thread) {                                                                                                         \
+			if (Thread::get_caller_id() == server_thread || RSG::texture_storage->can_create_resources_async()) {                    \
+				RSG::texture_storage->m_type##_initialize(ret, p1, p2, p3, p4, p5, p6);                                              \
+			} else {                                                                                                                 \
+				command_queue.push(RSG::texture_storage, &RendererTextureStorage::m_type##_initialize, ret, p1, p2, p3, p4, p5, p6); \
+			}                                                                                                                        \
+		} else {                                                                                                                     \
+			_THREAD_SAFE_METHOD_                                                                                                     \
```

Comment:
I don't think we need the mutex for texture creation functions. Notice how they previously never even flushed the command queue. When called from the server thread, the call should be able to go through immediately and without a mutex